### PR TITLE
8HRTuUnq: Validate Local Config Certificates

### PR DESCRIPTION
**Pull request by kerrr**
This is big(ish) refactor of how certificates from local config are validated. We want to be able to log potential issues with certificates at different log levels. It is also now expected that local certificates could be missing, because they will be overridden by self-service.

Review by commits

### CODE REVIEW

**Comment by kerrr** [comment]
I've fixed a couple of the codacy checks, but I'm going to leave the others for now.

File: hub/config/src/test/java/uk/gov/ida/hub/config/domain/CertificateChainConfigValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        when(certificateChainValidator.validate(signingCert.getX509Certificate().get(), trustStore)).thenReturn(CertificateValidity.valid());
+
+        validator.validate(Set.of(config));
+
+        List<ILoggingEvent> logsList = listAppender.list;
+        assertThat(logsList.get(0).getLevel().toString()).isEqualTo(Level.WARN.toString());
+        assertThat(logsList.get(0).getMessage()).contains("Bad Cert Chain");
+
+    }
+
```
Comment (alan-gds) [inline hub/config/src/test/java/uk/gov/ida/hub/config/domain/CertificateChainConfigValidatorTest.java]: Might be worth testing that we *don't* log anything when a cert is present and valid (both for self-service and for local).

File: hub/config/src/test/java/uk/gov/ida/hub/config/data/ConfigDataBootstrapTest.java [hunk 2]
```diff
@@ -155,9 +152,7 @@ public void continuesToStart_WhenCertificateCheckHasInvalidCertificates() {
         InvalidCertificateDto invalidIdpCertificateDto = new InvalidCertificateDto(idpEntityId, CertPathValidatorException.BasicReason.INVALID_SIGNATURE, CertificateUse.SIGNING, FederationEntityType.IDP, "certificate was bad!");
         InvalidCertificateDto invalidMatchingServiceCertificateDto = new InvalidCertificateDto(matchingServiceId, CertPathValidatorException.BasicReason.NOT_YET_VALID, CertificateUse.SIGNING, FederationEntityType.MS, "certificate was not yet valid!");
 
-        doThrow(createInvalidCertificatesException(ImmutableList.of(invalidMatchingServiceCertificateDto, invalidIdpCertificateDto))).when(certificateChainConfigValidator).validate(
-                ImmutableSet.of(transactionConfigData),
-                ImmutableSet.of(matchingServiceConfigData));
+        doThrow(createInvalidCertificatesException(ImmutableList.of(invalidMatchingServiceCertificateDto, invalidIdpCertificateDto))).when(certificateChainConfigValidator).validate(any());
```
[the remark this answers is not in the document]
Comment (kerrr) [reply hub/config/src/test/java/uk/gov/ida/hub/config/data/ConfigDataBootstrapTest.java]: The signature of `CertificateChainConfigValidator.validate()` changed and I was lazy when fixing it up. I've fixed it in a more specific way now.

File: hub/config/src/test/java/uk/gov/ida/hub/config/domain/CertificateChainConfigValidatorTest.java (hunk 1 above)
Comment (kerrr) [reply hub/config/src/test/java/uk/gov/ida/hub/config/domain/CertificateChainConfigValidatorTest.java]: Yup, good idea.

**Pull request merged**